lib/internal/streams/from.js

```diff
@@ -75,23 +79,37 @@ function from(Readable, iterable, opts) {
   }
 
   async function next() {
-    try {
-      const { value, done } = await iterator.next();
-      if (done) {
-        readable.push(null);
-      } else {
-        const res = await value;
-        if (res === null) {
-          reading = false;
-          throw new ERR_STREAM_NULL_VALUES();
-        } else if (readable.push(res)) {
-          next();
+    for (;;) {
+      try {
+        let r;
+        if (isAysnc) {
+          r = await iterator.next();
+        } else {
+          r = iterator.next();
+        }
+        const { value, done } = r;
+        if (done) {
+          readable.push(null);
         } else {
-          reading = false;
+          let res;
+          if (PromisePrototypeResovle(value) === value) {
```

Comment:
What does this do?

Reply:
Test if the value is Promise, if it isn't, it don't need to be awaited